iproute2: update to 6.9.0

rany2 opened the pull request:
## [iproute2: fix build on GCC 14](https://github.com/openwrt/openwrt/pull/15480/commits/f9d74048108a27be73aad72f8dcf38804ddbbf25)

Upstream patches:

401-bridge-vlan.c-bridge-vlan.c-fix-build-with-gcc-14-on.patch
402-bpf-fix-warning-from-basename.patch
403-bpf-include-libgen.h-for-basename.patch

The patch (400-rdma-include-libgen.h-for-basename.patch) was not
submitted upstream but just adds a missing include for basename.

## [iproute2: update to 6.9.0](https://github.com/openwrt/openwrt/pull/15480/commits/6f8af0eab969e97b58dc63e1ecf9d9558e49e5bb)

Support for iptables action has been dropped. Remove tc-mod-iptables and related
patch (175-reduce-dynamic-syms.patch).

We also add the missing libbpf dependency for `ss` since iproute 8740ca9
("ss: add support for BPF socket-local storage") now means that `ss` requires
libbpf as well.

Fix 170-ip_tiny.patch, as the help text didn't match all the included functions.

Drop upstreamed patches 402-bpf-fix-warning-from-basename.patch
and 403-bpf-include-libgen.h-for-basename.patch.

All other patches automatically rebased.

rany2 committed:
iproute2: fix build on GCC 14

Upstream patches:

401-bridge-vlan.c-bridge-vlan.c-fix-build-with-gcc-14-on.patch
402-bpf-fix-warning-from-basename.patch
403-bpf-include-libgen.h-for-basename.patch

The patch (400-rdma-include-libgen.h-for-basename.patch) was not
submitted upstream but just adds a missing include for basename.

Signed-off-by: Rany Hany <rany_hany@riseup.net>

neheb commented:
https://github.com/iproute2/iproute2/commit/69e3b2fadcd32683db2942f31fe41f0fbb2185f8 needs to be backported as well.

There are also time_t format errors. musl specific.

rany2 commented:
Done @neheb, thanks for the review!

neheb commented:
I'll also mention this x86 specific failure: https://github.com/openwrt/openwrt/pull/15418#issuecomment-2106455324

rany2 commented:
Oh, I did build test and it passed for arm; I'll try and x86 and let you know. I didn't check the thread for the PR that added support.

rany2 commented:
@neheb It seems like the error in rdma was the last issue I needed to address, can you confirm this?

neheb commented:
Can you add a patch updating to 6.9.0? https://github.com/openwrt/openwrt/pull/15418 seems stalled.

rany2 commented:
@neheb you mean you want this patch to also upgrade to 6.9.0? I'm willing to do so, just need a few hours

neheb commented:
Yeah. basically a second commit,

rany2 commented:
@neheb I can't build `ss` package with #15418, can you please confirm that you have that issue too before I just do a `dirclean`?

```
rm -rf /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/.pkgdir/ip-bridge.installed /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/.pkgdir/ip-bridge
mkdir -p /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/.pkgdir/ip-bridge
install -d -m0755 /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/.pkgdir/ip-bridge/usr/sbin
install -m0755 /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/bridge/bridge /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/.pkgdir/ip-bridge/usr/sbin/
touch /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/.pkgdir/ip-bridge.installed
rm -rf /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/ipkg-x86_64/ss
mkdir -p /home/rany/projects/openwrt/bin/targets/x86/64/packages /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/ipkg-x86_64/ss /home/rany/projects/openwrt/staging_dir/target-x86_64_musl/pkginfo
install -d -m0755 /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/ipkg-x86_64/ss/usr/sbin
install -m0755 /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/misc/ss /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/ipkg-x86_64/ss/usr/sbin/
find /home/rany/projects/openwrt/build_dir/target-x86_64_musl/linux-x86_64/iproute2-ipfull/iproute2-6.9.0/ipkg-x86_64/ss -name 'CVS' -o -name '.svn' -o -name '.#*' -o -name '*~'| xargs -r rm -rf
Package ss is missing dependencies for the following libraries:
libbpf.so.1
make[2]: *** [Makefile:275: /home/rany/projects/openwrt/bin/packages/x86_64/base/ss_6.9.0-r1_x86_64.ipk] Error 1
make[2]: Leaving directory '/home/rany/projects/openwrt/package/network/utils/iproute2'
time: package/network/utils/iproute2/ipfull/compile#0.85#0.24#5.65
    ERROR: package/network/utils/iproute2 failed to build (build variant: ipfull).
make[1]: *** [package/Makefile:167: package/network/utils/iproute2/compile] Error 1
make[1]: Leaving directory '/home/rany/projects/openwrt'
make: *** [/home/rany/projects/openwrt/include/toplevel.mk:233: package/network/utils/iproute2/compile] Error 2
nice -19 make -j1 package/network/utils/iproute2/compile V=s  37.46s user 6.11s system 85% cpu 50.846 total
```

neheb commented:
sounds like something needs disabling.

rany2 commented:
P.S. I think nothing was wrong on my end, it's just that ss now depends on libbpf: https://github.com/iproute2/iproute2/commit/8740ca9dcd3ccf1c75c362900cb3218ae3204cf5

I'll fix this issue with that update to 6.9.0 you linked to now and update my PR.

rsalvaterra commented:
> Can you add a patch updating to 6.9.0? #15418 seems stalled.

It's not stalled. You specifically mentioned it should be rebased on top of another pull request, so I'm waiting for it to get merged…

rany2 commented:
@rsalvaterra I was asked to add your patch to update to 6.9.0. So the scope of the PR changed from just fixing compilation on GCC14 to that and also updating to 6.9.0. I also fixed building the `ss` package on 6.9.0 as `ss` now depends on libbpf since 6.9.0.

rsalvaterra commented:
That's strange, I had no issues building `ss` with my pull request. Then again, I have a `libbpf` dependency on my builds for other reasons. Good catch.

hingbong commented:
applied patch and built failed with gcc 14
```
2024-05-25T07:45:44.4389217Z make[3]: Entering directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt/package/network/utils/iproute2'
2024-05-25T07:45:44.6254451Z rm -f /home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0/.built
2024-05-25T07:45:44.6286094Z touch /home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0/.built_check
2024-05-25T07:45:44.6332920Z CFLAGS="-O3 -pipe -mcpu=cortex-a53+crc+crypto+rdma -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0=iproute2-6.9.0 -ffunction-sections -fdata-sections -flto=auto -fno-fat-lto-objects -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro    -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/usr/include -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/include -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/include/fortify   -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/libnl-tiny " CXXFLAGS="-O3 -pipe -mcpu=cortex-a53+crc+crypto+rdma -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0=iproute2-6.9.0 -ffunction-sections -fdata-sections -flto=auto -fno-fat-lto-objects -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro   -ffunction-sections -fdata-sections -flto=auto -fno-fat-lto-objects  -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/usr/include -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/include -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/include/fortify   -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/libnl-tiny " LDFLAGS="-L/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/usr/lib -L/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/lib -fuse-ld=mold -Wl,--gc-sections -flto=auto -fuse-linker-plugin -znow -zrelro   -Wl,--as-needed " make  -C /home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0 AR="aarch64-openwrt-linux-musl-gcc-ar" AS="aarch64-openwrt-linux-musl-gcc -c -O3 -pipe -mcpu=cortex-a53+crc+crypto+rdma -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0=iproute2-6.9.0 -ffunction-sections -fdata-sections -flto=auto -fno-fat-lto-objects -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  " LD="aarch64-openwrt-linux-musl-ld.bfd" NM="aarch64-openwrt-linux-musl-gcc-nm" CC="aarch64-openwrt-linux-musl-gcc" GCC="aarch64-openwrt-linux-musl-gcc" CXX="aarch64-openwrt-linux-musl-g++" RANLIB="aarch64-openwrt-linux-musl-gcc-ranlib" STRIP=aarch64-openwrt-linux-musl-strip OBJCOPY=aarch64-openwrt-linux-musl-objcopy OBJDUMP=aarch64-openwrt-linux-musl-objdump SIZE=aarch64-openwrt-linux-musl-size CROSS="aarch64-openwrt-linux-musl-" ARCH="aarch64" KERNEL_INCLUDE="/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/linux-6.6.30/include/uapi" SHARED_LIBS= IP_CONFIG_TINY= BUILD_VARIANT=tctiny LIBBPF_FORCE=off HAVE_ELF= HAVE_MNL= HAVE_CAP= HAVE_TIRPC=n IPT_LIB_DIR=/usr/lib/iptables XT_LIB_DIR=/usr/lib/iptables TC_CONFIG_XT=n TC_CONFIG_XT_OLD=n TC_CONFIG_XT_OLD_H=n TC_CONFIG_IPSET=n FPIC="-DPIC -fPIC" V=''
2024-05-25T07:45:44.6352826Z make[4]: Entering directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0'
2024-05-25T07:45:44.6354346Z 
2024-05-25T07:45:44.6354487Z lib
2024-05-25T07:45:44.6380260Z make[5]: Entering directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0/lib'
2024-05-25T07:45:44.6382577Z make[5]: Nothing to be done for 'all'.
2024-05-25T07:45:44.6384222Z make[5]: Leaving directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0/lib'
2024-05-25T07:45:44.6385297Z 
2024-05-25T07:45:44.6385394Z ip
2024-05-25T07:45:44.6402218Z make[5]: Entering directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0/ip'
2024-05-25T07:45:44.6404081Z make[5]: Nothing to be done for 'all'.
2024-05-25T07:45:44.6406178Z make[5]: Leaving directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0/ip'
2024-05-25T07:45:44.6407726Z 
2024-05-25T07:45:44.6407864Z tc
2024-05-25T07:45:44.6430413Z make[5]: Entering directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0/tc'
2024-05-25T07:45:44.6439591Z aarch64-openwrt-linux-musl-gcc -Wall -Wstrict-prototypes  -Wmissing-prototypes -Wmissing-declarations -Wold-style-definition -Wformat=2 -O2 -pipe -I../include -I../include/uapi -DRESOLVE_HOSTNAMES -DLIBDIR=\"/usr/lib\" -DNO_SHARED_LIBS -DCONF_USR_DIR=\"/usr/share/iproute2\" -DCONF_ETC_DIR=\"/etc/iproute2\" -DNETNS_RUN_DIR=\"/var/run/netns\" -DNETNS_ETC_DIR=\"/etc/netns\" -DARPDDIR=\"/var/lib/arpd\" -DCONF_COLOR=COLOR_OPT_NEVER -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -O3 -pipe -mcpu=cortex-a53+crc+crypto+rdma -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0=iproute2-6.9.0 -ffunction-sections -fdata-sections -flto=auto -fno-fat-lto-objects -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro    -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/usr/include -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/include -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/include/fortify   -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/libnl-tiny  -DHAVE_SETNS -DHAVE_HANDLE_AT -DHAVE_SETNS -DHAVE_HANDLE_AT -DCONFIG_GACT -DCONFIG_GACT_PROB -DIPT_LIB_DIR=\"/usr/lib/iptables\" -DXT_LIB_DIR=\"/usr/lib/iptables\" -DYY_NO_INPUT   -c -o emp_ematch.tab.o emp_ematch.tab.c
2024-05-25T07:45:44.6953529Z flex  -oemp_ematch.lex.c emp_ematch.l
2024-05-25T07:45:44.7017922Z aarch64-openwrt-linux-musl-gcc -Wall -Wstrict-prototypes  -Wmissing-prototypes -Wmissing-declarations -Wold-style-definition -Wformat=2 -O2 -pipe -I../include -I../include/uapi -DRESOLVE_HOSTNAMES -DLIBDIR=\"/usr/lib\" -DNO_SHARED_LIBS -DCONF_USR_DIR=\"/usr/share/iproute2\" -DCONF_ETC_DIR=\"/etc/iproute2\" -DNETNS_RUN_DIR=\"/var/run/netns\" -DNETNS_ETC_DIR=\"/etc/netns\" -DARPDDIR=\"/var/lib/arpd\" -DCONF_COLOR=COLOR_OPT_NEVER -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -O3 -pipe -mcpu=cortex-a53+crc+crypto+rdma -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0=iproute2-6.9.0 -ffunction-sections -fdata-sections -flto=auto -fno-fat-lto-objects -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro    -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/usr/include -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/include -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/include/fortify   -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/libnl-tiny  -DHAVE_SETNS -DHAVE_HANDLE_AT -DHAVE_SETNS -DHAVE_HANDLE_AT -DCONFIG_GACT -DCONFIG_GACT_PROB -DIPT_LIB_DIR=\"/usr/lib/iptables\" -DXT_LIB_DIR=\"/usr/lib/iptables\" -DYY_NO_INPUT   -c -o emp_ematch.lex.o emp_ematch.lex.c
2024-05-25T07:45:44.7745424Z aarch64-openwrt-linux-musl-gcc -Wall -Wstrict-prototypes  -Wmissing-prototypes -Wmissing-declarations -Wold-style-definition -Wformat=2 -O2 -pipe -I../include -I../include/uapi -DRESOLVE_HOSTNAMES -DLIBDIR=\"/usr/lib\" -DNO_SHARED_LIBS -DCONF_USR_DIR=\"/usr/share/iproute2\" -DCONF_ETC_DIR=\"/etc/iproute2\" -DNETNS_RUN_DIR=\"/var/run/netns\" -DNETNS_ETC_DIR=\"/etc/netns\" -DARPDDIR=\"/var/lib/arpd\" -DCONF_COLOR=COLOR_OPT_NEVER -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -O3 -pipe -mcpu=cortex-a53+crc+crypto+rdma -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0=iproute2-6.9.0 -ffunction-sections -fdata-sections -flto=auto -fno-fat-lto-objects -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro    -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/usr/include -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/include -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_musl/include/fortify   -I/home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/include/libnl-tiny  -DHAVE_SETNS -DHAVE_HANDLE_AT -DHAVE_SETNS -DHAVE_HANDLE_AT -DCONFIG_GACT -DCONFIG_GACT_PROB -DIPT_LIB_DIR=\"/usr/lib/iptables\" -DXT_LIB_DIR=\"/usr/lib/iptables\" -DYY_NO_INPUT   -c -o m_nssmirred.o m_nssmirred.c
2024-05-25T07:45:44.8013105Z m_nssmirred.c:181:23: error: initialization of 'int (*)(const struct action_util *, int *, char ***, int,  struct nlmsghdr *)' from incompatible pointer type 'int (*)(struct action_util *, int *, char ***, int,  struct nlmsghdr *)' [-Wincompatible-pointer-types]
2024-05-25T07:45:44.8015476Z   181 |         .parse_aopt = parse_nss_mirred,
2024-05-25T07:45:44.8016123Z       |                       ^~~~~~~~~~~~~~~~
2024-05-25T07:45:44.8017192Z m_nssmirred.c:181:23: note: (near initialization for 'nssmirred_action_util.parse_aopt')
2024-05-25T07:45:44.8019719Z m_nssmirred.c:182:23: error: initialization of 'int (*)(const struct action_util *, FILE *, struct rtattr *)' from incompatible pointer type 'int (*)(struct action_util *, FILE *, struct rtattr *)' [-Wincompatible-pointer-types]
2024-05-25T07:45:44.8021592Z   182 |         .print_aopt = print_nss_mirred,
2024-05-25T07:45:44.8022197Z       |                       ^~~~~~~~~~~~~~~~
2024-05-25T07:45:44.8023194Z m_nssmirred.c:182:23: note: (near initialization for 'nssmirred_action_util.print_aopt')
2024-05-25T07:45:44.8042916Z rm emp_ematch.lex.c
2024-05-25T07:45:44.8044572Z make[5]: Leaving directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0/tc'
2024-05-25T07:45:44.8045852Z make[5]: *** [../config.mk:33: m_nssmirred.o] Error 1
2024-05-25T07:45:44.8046704Z make[4]: *** [Makefile:81: all] Error 2
2024-05-25T07:45:44.8048354Z make[4]: Leaving directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0'
2024-05-25T07:45:44.8051337Z make[3]: *** [Makefile:261: /home/runner/work/ax3600-builder/ax3600-builder/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq807x/iproute2-tctiny/iproute2-6.9.0/.built] Error 2
2024-05-25T07:45:44.8053747Z make[3]: Leaving directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt/package/network/utils/iproute2'
2024-05-25T07:45:44.8055790Z time: package/network/utils/iproute2/tctiny/compile#0.30#0.09#0.37
2024-05-25T07:45:44.8063687Z     ERROR: package/network/utils/iproute2 failed to build (build variant: tctiny).
2024-05-25T07:45:44.8067625Z make[2]: *** [package/Makefile:173: package/network/utils/iproute2/compile] Error 1
2024-05-25T07:45:44.8072609Z make[2]: Leaving directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt'
2024-05-25T07:45:44.8080706Z make[1]: *** [package/Makefile:167: /home/runner/work/ax3600-builder/ax3600-builder/openwrt/staging_dir/target-aarch64_cortex-a53_musl/stamp/.package_compile] Error 2
2024-05-25T07:45:44.8092676Z make[1]: Leaving directory '/home/runner/work/ax3600-builder/ax3600-builder/openwrt'
2024-05-25T07:45:44.8127422Z make: *** [/home/runner/work/ax3600-builder/ax3600-builder/openwrt/include/toplevel.mk:233: world] Error 2
```

rany2 commented:
@hingbong Can you upload your .config?

hingbong commented:
> @hingbong Can you upload your .config?

here is the defconfig
```
CONFIG_TARGET_qualcommax=y
CONFIG_TARGET_qualcommax_ipq807x=y
CONFIG_TARGET_qualcommax_ipq807x_DEVICE_xiaomi_ax3600=y
CONFIG_DEVEL=y
CONFIG_TOOLCHAINOPTS=y
# CONFIG_ATH11K_MEM_PROFILE_1G is not set
CONFIG_ATH11K_MEM_PROFILE_512M=y
CONFIG_ATH11K_NSS_SUPPORT=y
CONFIG_BIND_ENABLE_DOH=y
CONFIG_BUILD_PATENTED=y
CONFIG_GCC_DEFAULT_PIE=y
CONFIG_GCC_USE_GRAPHITE=y
# CONFIG_GCC_USE_VERSION_13 is not set
CONFIG_GCC_USE_VERSION_14=y
CONFIG_GCC_VERSION="14.1.0"
CONFIG_GCC_VERSION_14=y
CONFIG_KERNEL_ARM_PMU=y
CONFIG_KERNEL_ARM_PMUV3=y
CONFIG_KERNEL_CFQ_GROUP_IOSCHED=y
CONFIG_KERNEL_PERF_EVENTS=y
# CONFIG_KERNEL_WERROR is not set
CONFIG_LIBCURL_COOKIES=y
CONFIG_LIBCURL_CRYPTO_AUTH=y
CONFIG_LIBCURL_DICT=y
CONFIG_LIBCURL_FILE=y
CONFIG_LIBCURL_FTP=y
CONFIG_LIBCURL_GOPHER=y
CONFIG_LIBCURL_HTTP=y
CONFIG_LIBCURL_IMAP=y
CONFIG_LIBCURL_LIBCURL_OPTION=y
CONFIG_LIBCURL_LIBIDN2=y
CONFIG_LIBCURL_NGHTTP2=y
CONFIG_LIBCURL_NTLM=y
CONFIG_LIBCURL_OPENSSL=y
CONFIG_LIBCURL_POP3=y
CONFIG_LIBCURL_PROXY=y
CONFIG_LIBCURL_RTSP=y
CONFIG_LIBCURL_SMB=y
CONFIG_LIBCURL_SMTP=y
CONFIG_LIBCURL_SSH2=y
CONFIG_LIBCURL_TELNET=y
CONFIG_LIBCURL_TFTP=y
CONFIG_LIBCURL_THREADED_RESOLVER=y
CONFIG_LIBCURL_TLS_SRP=y
CONFIG_LIBCURL_UNIX_SOCKETS=y
CONFIG_LIBCURL_VERBOSE=y
CONFIG_LIBCURL_ZLIB=y
CONFIG_LIBCURL_ZSTD=y
CONFIG_LIBSSH2_OPENSSL=y
CONFIG_MOLD=y
CONFIG_NGINX_HTTP_ACCESS=y
CONFIG_NGINX_HTTP_AUTH_BASIC=y
CONFIG_NGINX_HTTP_AUTOINDEX=y
CONFIG_NGINX_HTTP_BROWSER=y
CONFIG_NGINX_HTTP_CACHE=y
CONFIG_NGINX_HTTP_CHARSET=y
CONFIG_NGINX_HTTP_EMPTY_GIF=y
CONFIG_NGINX_HTTP_FASTCGI=y
CONFIG_NGINX_HTTP_GEO=y
CONFIG_NGINX_HTTP_GZIP=y
CONFIG_NGINX_HTTP_GZIP_STATIC=y
CONFIG_NGINX_HTTP_LIMIT_CONN=y
CONFIG_NGINX_HTTP_LIMIT_REQ=y
CONFIG_NGINX_HTTP_MAP=y
CONFIG_NGINX_HTTP_MEMCACHED=y
CONFIG_NGINX_HTTP_PROXY=y
CONFIG_NGINX_HTTP_QUIC=y
CONFIG_NGINX_HTTP_REFERER=y
CONFIG_NGINX_HTTP_REWRITE=y
CONFIG_NGINX_HTTP_SCGI=y
CONFIG_NGINX_HTTP_SPLIT_CLIENTS=y
CONFIG_NGINX_HTTP_SSI=y
CONFIG_NGINX_HTTP_UPSTREAM_HASH=y
CONFIG_NGINX_HTTP_UPSTREAM_IP_HASH=y
CONFIG_NGINX_HTTP_UPSTREAM_KEEPALIVE=y
CONFIG_NGINX_HTTP_UPSTREAM_LEAST_CONN=y
CONFIG_NGINX_HTTP_USERID=y
CONFIG_NGINX_HTTP_UWSGI=y
CONFIG_NGINX_HTTP_V2=y
CONFIG_NGINX_PCRE=y
CONFIG_NSS_DRV_BRIDGE_ENABLE=y
CONFIG_NSS_DRV_C2C_ENABLE=y
CONFIG_NSS_DRV_CAPWAP_ENABLE=y
CONFIG_NSS_DRV_CLMAP_ENABLE=y
CONFIG_NSS_DRV_CRYPTO_ENABLE=y
CONFIG_NSS_DRV_DTLS_ENABLE=y
CONFIG_NSS_DRV_GRE_ENABLE=y
CONFIG_NSS_DRV_GRE_REDIR_ENABLE=y
CONFIG_NSS_DRV_IGS_ENABLE=y
CONFIG_NSS_DRV_IPSEC_ENABLE=y
CONFIG_NSS_DRV_IPV4_REASM_ENABLE=y
CONFIG_NSS_DRV_IPV6_ENABLE=y
CONFIG_NSS_DRV_IPV6_REASM_ENABLE=y
CONFIG_NSS_DRV_L2TP_ENABLE=y
CONFIG_NSS_DRV_LSO_RX_ENABLE=y
CONFIG_NSS_DRV_MAPT_ENABLE=y
CONFIG_NSS_DRV_MATCH_ENABLE=y
CONFIG_NSS_DRV_MIRROR_ENABLE=y
CONFIG_NSS_DRV_OAM_ENABLE=y
CONFIG_NSS_DRV_PPPOE_ENABLE=y
CONFIG_NSS_DRV_PPTP_ENABLE=y
CONFIG_NSS_DRV_PVXLAN_ENABLE=y
CONFIG_NSS_DRV_QRFS_ENABLE=y
CONFIG_NSS_DRV_RMNET_ENABLE=y
CONFIG_NSS_DRV_SHAPER_ENABLE=y
CONFIG_NSS_DRV_TLS_ENABLE=y
CONFIG_NSS_DRV_UDP_ST_ENABLE=y
CONFIG_NSS_DRV_VIRT_IF_ENABLE=y
CONFIG_NSS_DRV_VLAN_ENABLE=y
CONFIG_NSS_DRV_VXLAN_ENABLE=y
CONFIG_NSS_DRV_WIFIOFFLOAD_ENABLE=y
CONFIG_NSS_DRV_WIFI_EXT_VDEV_ENABLE=y
CONFIG_NSS_DRV_WIFI_LEGACY_ENABLE=y
CONFIG_NSS_MEM_PROFILE_MEDIUM=y
CONFIG_OPENSSH_LIBFIDO2=y
CONFIG_OPENSSL_ENGINE=y
CONFIG_OPENSSL_ENGINE_BUILTIN=y
CONFIG_OPENSSL_OPTIMIZE_SPEED=y
CONFIG_OPENSSL_WITH_ARIA=y
CONFIG_OPENSSL_WITH_ASM=y
CONFIG_OPENSSL_WITH_BLAKE2=y
CONFIG_OPENSSL_WITH_CAMELLIA=y
CONFIG_OPENSSL_WITH_CHACHA_POLY1305=y
CONFIG_OPENSSL_WITH_CMS=y
CONFIG_OPENSSL_WITH_COMPRESSION=y
CONFIG_OPENSSL_WITH_DEPRECATED=y
CONFIG_OPENSSL_WITH_DTLS=y
CONFIG_OPENSSL_WITH_EC2M=y
CONFIG_OPENSSL_WITH_ERROR_MESSAGES=y
CONFIG_OPENSSL_WITH_IDEA=y
CONFIG_OPENSSL_WITH_MDC2=y
CONFIG_OPENSSL_WITH_NPN=y
CONFIG_OPENSSL_WITH_PSK=y
CONFIG_OPENSSL_WITH_RFC3779=y
CONFIG_OPENSSL_WITH_SEED=y
CONFIG_OPENSSL_WITH_SM234=y
CONFIG_OPENSSL_WITH_SRP=y
CONFIG_OPENSSL_WITH_TLS13=y
CONFIG_OPENSSL_WITH_WHIRLPOOL=y
CONFIG_PACKAGE_MAC80211_NSS_SUPPORT=y
CONFIG_PACKAGE_bind-dig=y
CONFIG_PACKAGE_bind-libs=y
CONFIG_PACKAGE_btop=y
CONFIG_PACKAGE_cgi-io=y
CONFIG_PACKAGE_curl=y
# CONFIG_PACKAGE_dnsmasq is not set
# CONFIG_PACKAGE_dropbear is not set
CONFIG_PACKAGE_ethtool=y
CONFIG_PACKAGE_hickory-dns=y
CONFIG_PACKAGE_iperf3=y
# CONFIG_PACKAGE_iw is not set
CONFIG_PACKAGE_iw-full=y
CONFIG_PACKAGE_kmod-crypto-authenc=y
# CONFIG_PACKAGE_kmod-fs-ext4 is not set
CONFIG_PACKAGE_kmod-gre=y
CONFIG_PACKAGE_kmod-gre6=y
CONFIG_PACKAGE_kmod-ifb=y
CONFIG_PACKAGE_kmod-ip6-tunnel=y
CONFIG_PACKAGE_kmod-iptunnel=y
CONFIG_PACKAGE_kmod-iptunnel6=y
CONFIG_PACKAGE_kmod-l2tp=y
# CONFIG_PACKAGE_kmod-lib-crc16 is not set
CONFIG_PACKAGE_kmod-nat46=y
CONFIG_PACKAGE_kmod-nf-conntrack-netlink=y
CONFIG_PACKAGE_kmod-nf-ipt=y
CONFIG_PACKAGE_kmod-nft-compat=y
CONFIG_PACKAGE_kmod-pppol2tp=y
CONFIG_PACKAGE_kmod-pptp=y
CONFIG_PACKAGE_kmod-qca-mcs=y
CONFIG_PACKAGE_kmod-qca-nss-drv=y
CONFIG_PACKAGE_kmod-qca-nss-drv-pppoe=y
CONFIG_PACKAGE_kmod-qca-nss-drv-vlan-mgr=y
CONFIG_PACKAGE_kmod-qca-nss-ecm=y
CONFIG_PACKAGE_kmod-sched-cake=y
CONFIG_PACKAGE_kmod-sched-core=y
CONFIG_PACKAGE_kmod-tcp-bbr=y
CONFIG_PACKAGE_kmod-udptunnel4=y
CONFIG_PACKAGE_kmod-udptunnel6=y
CONFIG_PACKAGE_kmod-vxlan=y
# CONFIG_PACKAGE_knot-resolver_dnstap is not set
CONFIG_PACKAGE_libatomic=y
CONFIG_PACKAGE_libcap=y
CONFIG_PACKAGE_libcbor=y
CONFIG_PACKAGE_libcurl=y
CONFIG_PACKAGE_libevdev=y
CONFIG_PACKAGE_libevent2=y
CONFIG_PACKAGE_libfido2=y
CONFIG_PACKAGE_libidn2=y
CONFIG_PACKAGE_libiperf3=y
CONFIG_PACKAGE_liblua=y
CONFIG_PACKAGE_liblucihttp=y
CONFIG_PACKAGE_liblucihttp-lua=y
CONFIG_PACKAGE_liblucihttp-ucode=y
# CONFIG_PACKAGE_libmbedtls is not set
CONFIG_PACKAGE_libnghttp2=y
CONFIG_PACKAGE_libnghttp3=y
CONFIG_PACKAGE_libngtcp2=y
CONFIG_PACKAGE_libopenssl=y
CONFIG_PACKAGE_libopenssl-conf=y
CONFIG_PACKAGE_libopenssl-legacy=y
CONFIG_PACKAGE_libpcre2=y
CONFIG_PACKAGE_libssh2=y
CONFIG_PACKAGE_libstdcpp=y
CONFIG_PACKAGE_libtirpc=y
CONFIG_PACKAGE_libubus-lua=y
CONFIG_PACKAGE_libudev-zero=y
CONFIG_PACKAGE_libunbound=y
CONFIG_PACKAGE_libunbound_ipset=y
CONFIG_PACKAGE_libunbound_libevent=y
CONFIG_PACKAGE_libunbound_libpthread=y
CONFIG_PACKAGE_libunbound_nghttp2=y
CONFIG_PACKAGE_libunistring=y
# CONFIG_PACKAGE_libustream-mbedtls is not set
CONFIG_PACKAGE_libustream-openssl=y
CONFIG_PACKAGE_libuv=y
CONFIG_PACKAGE_libzstd=y
CONFIG_PACKAGE_lsof=y
CONFIG_PACKAGE_lua=y
CONFIG_PACKAGE_luci-app-firewall=y
CONFIG_PACKAGE_luci-app-opkg=y
CONFIG_PACKAGE_luci-app-unbound=y
CONFIG_PACKAGE_luci-base=y
CONFIG_PACKAGE_luci-compat=y
CONFIG_PACKAGE_luci-lib-base=y
CONFIG_PACKAGE_luci-lib-ip=y
CONFIG_PACKAGE_luci-lib-jsonc=y
CONFIG_PACKAGE_luci-lib-nixio=y
CONFIG_PACKAGE_luci-lua-runtime=y
CONFIG_PACKAGE_luci-mod-admin-full=y
CONFIG_PACKAGE_luci-mod-network=y
CONFIG_PACKAGE_luci-mod-status=y
CONFIG_PACKAGE_luci-mod-system=y
CONFIG_PACKAGE_luci-nginx=y
CONFIG_PACKAGE_luci-proto-ipv6=y
CONFIG_PACKAGE_luci-proto-ppp=y
CONFIG_PACKAGE_luci-theme-bootstrap=y
CONFIG_PACKAGE_luci-theme-material=y
CONFIG_PACKAGE_nginx=y
CONFIG_PACKAGE_nginx-mod-luci=y
CONFIG_PACKAGE_nginx-mod-ubus=y
CONFIG_PACKAGE_nginx-ssl=y
CONFIG_PACKAGE_nginx-ssl-util=y
CONFIG_PACKAGE_nginx-util=y
CONFIG_PACKAGE_nss-firmware-default=y
CONFIG_PACKAGE_nss-firmware-ipq8074=y
CONFIG_PACKAGE_odhcpd=y
# CONFIG_PACKAGE_odhcpd-ipv6only is not set
CONFIG_PACKAGE_odhcpd_full_ext_cer_id=0
CONFIG_PACKAGE_openssh-keygen=y
CONFIG_PACKAGE_openssh-moduli=y
CONFIG_PACKAGE_openssh-server=y
CONFIG_PACKAGE_openssh-sftp-server=y
CONFIG_PACKAGE_pppoe-discovery=y
CONFIG_PACKAGE_rpcd=y
CONFIG_PACKAGE_rpcd-mod-file=y
CONFIG_PACKAGE_rpcd-mod-iwinfo=y
CONFIG_PACKAGE_rpcd-mod-luci=y
CONFIG_PACKAGE_rpcd-mod-rrdns=y
CONFIG_PACKAGE_rpcd-mod-ucode=y
CONFIG_PACKAGE_tc-tiny=y
CONFIG_PACKAGE_ucode-mod-html=y
CONFIG_PACKAGE_ucode-mod-lua=y
CONFIG_PACKAGE_ucode-mod-math=y
CONFIG_PACKAGE_unbound-control=y
CONFIG_PACKAGE_unbound-daemon=y
CONFIG_PACKAGE_uwsgi=y
CONFIG_PACKAGE_uwsgi-cgi-plugin=y
CONFIG_PACKAGE_uwsgi-luci-support=y
CONFIG_PACKAGE_uwsgi-syslog-plugin=y
CONFIG_PACKAGE_wireless-tools=y
# CONFIG_PACKAGE_wpad-basic-mbedtls is not set
CONFIG_PACKAGE_wpad-openssl=y
CONFIG_PACKAGE_zlib=y
CONFIG_PACKAGE_zoneinfo-asia=y
CONFIG_PACKAGE_zoneinfo-core=y
CONFIG_TARGET_OPTIMIZATION="-O3 -pipe -mcpu=cortex-a53+crc+crypto+rdma"
CONFIG_TARGET_OPTIONS=y
CONFIG_USE_GC_SECTIONS=y
CONFIG_USE_LTO=y
CONFIG_USE_MOLD=y
# CONFIG_WOLFSSL_ALT_NAMES is not set
# CONFIG_WOLFSSL_HAS_AES_CCM is not set
# CONFIG_WOLFSSL_HAS_ARC4 is not set
# CONFIG_WOLFSSL_HAS_CERTGEN is not set
# CONFIG_WOLFSSL_HAS_CHACHA_POLY is not set
# CONFIG_WOLFSSL_HAS_DH is not set
# CONFIG_WOLFSSL_HAS_ECC25519 is not set
# CONFIG_WOLFSSL_HAS_OCSP is not set
# CONFIG_WOLFSSL_HAS_OPENVPN is not set
# CONFIG_WOLFSSL_HAS_SESSION_TICKET is not set
# CONFIG_WOLFSSL_HAS_TLSV10 is not set
# CONFIG_WOLFSSL_HAS_TLSV13 is not set
# CONFIG_WOLFSSL_HAS_WPAS is not set
CONFIG_ZLIB_OPTIMIZE_SPEED=y
CONFIG_ZSTD_OPTIMIZE_O3=y
CONFIG_MOLD=y
CONFIG_USE_MOLD=y
```

rany2 commented:
@hingbong Sorry I don't have such a problem, I just did a completely clean build (distclean) and it built just fine.

rsalvaterra committed:
iproute2: update to 6.9.0

Support for iptables action has been dropped. Remove tc-mod-iptables and related
patch (175-reduce-dynamic-syms.patch).

We also add the missing libbpf dependency for `ss` since iproute 8740ca9
("ss: add support for BPF socket-local storage") now means that `ss` requires
libbpf as well.

Fix 170-ip_tiny.patch, as the help text didn't match all the included functions.

Drop upstreamed patches 402-bpf-fix-warning-from-basename.patch
and 403-bpf-include-libgen.h-for-basename.patch.

All other patches automatically rebased.

Co-authored-by: Rany Hany <rany_hany@riseup.net>
Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>
Signed-off-by: Rany Hany <rany_hany@riseup.net>

rany2 commented:
I updated this PR to use upstreamed version of the patch "bridge/vlan.c: bridge/vlan.c: fix build with gcc 14 on musl system".

httpstorm commented:
@rany2 
On a `frr` PR https://github.com/openwrt/packages/pull/24242#issuecomment-2134950553 of mine, @neheb pointed me to check whether the parameter to `basename` is writable. Here `cfg->object` is not writable: `const char *object;`. According to [1]
```
There are two different versions of basename() - the POSIX
       version described above, and the GNU version, which one gets
       after

               #define _GNU_SOURCE         /* See feature_test_macros(7) */
               #include <string.h>

       The GNU version never modifies its argument, and returns the
       empty string when path has a trailing slash, and in particular
       also when it is "/".  There is no GNU version of dirname().

       With glibc, one gets the POSIX version of basename() when
       <libgen.h> is included, and the GNU version otherwise.
```

[1] https://man7.org/linux/man-pages/man3/basename.3.html

httpstorm commented:
@rany2 
Yes, it is addressed in your PR. I'm sorry for the confusion and thanks for the help.

httpstorm commented:
@rany2 
Build tested: WRT3200ACM, host: macOS 14.5. If you provide any specific commands, I can also test run.

rany2 commented:
> Build tested: WRT3200ACM, host: macOS 14.5. If you provide any specific commands, I can also test run.

I've been using this for about a week now too so I think it's fine. I'm not really sure what I should do to get this merged. If you want you could test it out and if all looks good I'll add you in the Tested-by field.

httpstorm commented:
@rany2 
I enable it as part of the build process for my routers. It shows help when I run it. I've never ran any commands manually. I'm not sure if it is used in the background or not. If not, I do not qualify for the Tested-by field. Unless you know any specific commands, that can test your changes, and I run them.

rany2 commented:
> Unless you know any specific commands, that can test your changes, and I run them.

I guess there's no need, my changes just involve adding missing includes and dependencies.
